resize-observer/devicepixel2.html

```diff
@@ -0,0 +1,163 @@
+same as the body so they should both have a devicePixel position of 0,0
+
+It then sets a repeating background-image and background-size on the the
+body element and all of the .outer children elements so they have a 4x4
+checkerboard image at a size that should draw each pixel of the 4x4 
+checkerboard pattern in single devicePixels.
+
+The devicePixelContentBoxSize of all the elements is queried with
+a ResizeObserver.
+
+Finally the background-position of each child element is set so that
+the patterns should align.
+
+In other words
+
+  <div class"outer">
+    <div></div><div></div><div></div>
+  </div>
+
+The first child will be displayed at left = 0 so its background-position is
+set to 0px 0px. The second child should be displayed at 
+
+secondChildLeft = entry(firstChild).devicePixelContentBoxSize[0].inlineSize. 
+
+The 3rd child should be displayed at 
+
+thirdChildLeft = entry(firstChild).devicePixelContentBoxSize[0].inlineSize + 
+                 entry(secondChild).devicePixelContentBoxSize[0].inlineSize
+
+If the element's left edge mod 4 is an non-zero value then we need to set the
+background-position of the element's pattern left edge mod 4 devicePixels.
+
+If what devicePixelContentBox reports is correct the children's patterns
+will align with the pattern in the body and the children will be invisible.
+If what devicePixelContentBox reports is incorrect the patterns will not
+align and it should be clearly visible.
+*/
+
+const frontElem = document.querySelector('#front');
+
+const elemToDevicePixelSize = new Map();
+
+function setPatternsUsingSizeInfo(entries) {
+  for (const entry of entries) {
+    elemToDevicePixelSize.set(entry.target, {
+      inlineSize: entry.devicePixelContentBoxSize[0].inlineSize,
+      blockSize: entry.devicePixelContentBoxSize[0].blockSize,
+    });
+  }
+
+  let devicePixelY = 0;
+  for (const outerElem of frontElem.children) {
+    let devicePixelX = 0;
+    for (const childElem of outerElem.children) {
+      setPattern(childElem, devicePixelX, devicePixelY);
+      const devicePixelSize = elemToDevicePixelSize.get(childElem);
+      devicePixelX += devicePixelSize.inlineSize;
+    }
+
+    const outerEntry = elemToDevicePixelSize.get(outerElem);
+    const outerHeight = outerEntry.blockSize;
+    devicePixelY += outerHeight;
+  }
+
+  setPattern(document.body, 0, 0);
+}
+
+const observer = new ResizeObserver(setPatternsUsingSizeInfo);
+observer.observe(document.body, {box:"device-pixel-content-box"});
+for (let numFlexItems = 2; numFlexItems < 15; ++numFlexItems) {
+  const outer = document.createElement('div');
+  outer.className = 'outer';
+  observer.observe(outer, {box:"device-pixel-content-box"});
+  for (let j = 0; j < numFlexItems; ++j) {
+    const inner = document.createElement('div');
+    outer.appendChild(inner);
+    observer.observe(inner, {box:"device-pixel-content-box"});
+  }
+  frontElem.appendChild(outer);
+}
+</script>
```

Comment:
nit: add newline at end of file

(I think you fixed this for the reference case but not for the testcase)

Reply:
done